_examples/basic/5-cqrs-protobuf/main.go

```diff
@@ -207,56 +207,149 @@ func main() {
+			start := time.Now()
+
+			err := params.Handler.Handle(params.Message.Context(), params.Command)
+
+			logger.Info("Command handled", watermill.LogFields{
+				"command_name": params.CommandName,
+				"duration":     time.Since(start),
+				"err":          err,
+			})
+
+			return err
+		},
+		Marshaler: cqrsMarshaler,
+		Logger:    logger,
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	eventBus, err := cqrs.NewEventBusWithConfig(eventsPublisher, cqrs.EventBusConfig{
+		GeneratePublishTopic: func(params cqrs.GenerateEventPublishTopicParams) (string, error) {
 			// because we are using PubSub RabbitMQ config, we can use one topic for all events
-			return "events"
+			return "events", nil
 
 			// we can also use topic per event type
-			// return eventName
+			// return params.EventName, nil
+		},
+
+		OnPublish: func(params cqrs.OnEventSendParams) error {
+			logger.Info("Publishing event", watermill.LogFields{
+				"event_name": params.EventName,
+			})
+
+			params.Message.Metadata.Set("published_at", time.Now().String())
+
+			return nil
 		},
-		EventHandlers: func(cb *cqrs.CommandBus, eb *cqrs.EventBus) []cqrs.EventHandler {
-			return []cqrs.EventHandler{
-				OrderBeerOnRoomBooked{cb},
-				NewBookingsFinancialReport(),
-			}
+
+		Marshaler: cqrsMarshaler,
+		Logger:    logger,
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	commandProcessor.AddHandler(
+		BookRoomHandler{eventBus},
+		OrderBeerHandler{eventBus},
+	)
+
+	err = commandProcessor.AddHandlersToRouter(router)
+	if err != nil {
+		panic(err)
+	}
+
+	eventProcessor, err := cqrs.NewEventGroupProcessorWithConfig(cqrs.EventGroupProcessorConfig{
+		GenerateSubscribeTopic: func(params cqrs.EventGroupProcessorGenerateSubscribeTopicParams) (string, error) {
+			return "events", nil
 		},
-		EventsPublisher: eventsPublisher,
-		EventsSubscriberConstructor: func(handlerName string) (message.Subscriber, error) {
+		SubscriberConstructor: func(params cqrs.EventGroupProcessorSubscriberConstructorParams) (message.Subscriber, error) {
 			config := amqp.NewDurablePubSubConfig(
 				amqpAddress,
-				amqp.GenerateQueueNameTopicNameWithSuffix(handlerName),
+				amqp.GenerateQueueNameTopicNameWithSuffix(params.EventGroupName),
 			)
 
 			return amqp.NewSubscriber(config, logger)
 		},
-		Router:                router,
-		CommandEventMarshaler: cqrsMarshaler,
-		Logger:                logger,
+
+		OnHandle: func(params cqrs.EventGroupProcessorOnHandleParams) error {
+			start := time.Now()
+
+			err := params.Handler.Handle(params.Message.Context(), params.Event)
+
+			logger.Info("Event handled", watermill.LogFields{
+				"event_name": params.EventName,
+				"duration":   time.Since(start),
+				"err":        err,
+			})
+
+			return err
+		},
+
+		Marshaler: cqrsMarshaler,
+		Logger:    logger,
 	})
 	if err != nil {
 		panic(err)
 	}
 
+	err = eventProcessor.AddHandlersGroup(
+		"events",
+		OrderBeerOnRoomBooked{commandBus},
```

Comment:
Should all of these be in the same consumer group? 🤔

Reply:
in theory it doesn't - but we'll need to enable `AckOnUnknownEvent` (this was enabled by default earlier) because it will deliver all events to all handlers